### PR TITLE
Feedback

### PR DESCRIPTION
**Pull request by grebe**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -17,3 +17,5 @@ name := "rocketdsptools"
 organization := "edu.berkeley.cs"
 version := "0.1-SNAPSHOT"
 scalaVersion := "2.12.6"
+
+libraryDependencies += "edu.berkeley.cs" %% "rocket-dsptools" % "1.3-SNAPSHOT"
```
Comment (grebe) [inline build.sbt:21]: Oops, I left in my build system stuff. This shouldn't be needed by you

File: project/plugins.sbt [hunk 2]
```diff
@@ -0,0 +1 @@
+addSbtPlugin("org.scalastyle" %% "scalastyle-sbt-plugin" % "1.0.0")
```
Comment (grebe) [inline project/plugins.sbt:1]: I added a linter plugin

File: src/main/scala/GenericFIR.scala [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+import chisel3.util._
+import dspblocks._
+import dsptools.numbers._
+import freechips.rocketchip.amba.axi4stream._
+import freechips.rocketchip.config.Parameters
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.subsystem._
+
+class GenericFIRCellBundle[T<:Data:Ring](genIn:T, genOut:T) extends Bundle {
```
Comment (grebe) [inline src/main/scala/GenericFIR.scala:16]: Convention is to make classes capital in most cases

File: src/main/scala/GenericFIR.scala [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  // Construct a vector of GenericFIRDirectCells
+  val directCells = Seq.fill(coeffs.length){ Module(new GenericFIRDirectCell(genIn, genOut)).io }
+
+  // connect coefficients
+  for ((cell, coeff) <- directCells.zip(coeffs)) {
+    cell.coeff := coeff
+  }
+
+  // connect input to first cell.
+  directCells(0).in.bits.data := io.in.bits
```
Comment (grebe) [inline src/main/scala/GenericFIR.scala:65]: These should probably be head

File: src/main/scala/GenericFIR.scala [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+class GenericFIRDirectCell[T<:Data:Ring](genIn: T, genOut: T) extends Module {
+  val io = IO(GenericFIRCellIO(genIn, genOut))
+
+  // Passthrough ready
+  io.in.ready := io.out.ready
+
+  // TODO reason about ready/valid
+  // Describe what's going on
+  val validReg = RegNext(io.in.fire(), init = false.B)
+  io.out.valid := validReg
```
Comment (grebe) [inline src/main/scala/GenericFIR.scala:100]: I'm not sure that my logical changes are right, you should make sure ready/valid work

File: src/main/scala/Passthrough.scala [hunk 6]
```diff
@@ -10,7 +10,6 @@ import dsptools.numbers._
 import freechips.rocketchip.amba.axi4stream._
 import freechips.rocketchip.config.Parameters
 import freechips.rocketchip.diplomacy._
-import freechips.rocketchip.regmapper._
```
Comment (grebe) [inline src/main/scala/Passthrough.scala:13]: I got some warnings about unused imports, you should make sure they're clean

**Review by ryan-lund — APPROVED** [review]

**Pull request merged**